src/Dualize/Program.hs

```diff
@@ -24,8 +26,9 @@ dualDataDecl (RST.RefinementDecl loc doc rntn dc pk (sigsPos,sigsNeg)) =
 dualXtorSig ::  PolarityRep pol -> RST.XtorSig pol -> RST.XtorSig pol 
 dualXtorSig pol (RST.MkXtorSig xtor lctx) = RST.MkXtorSig (dualXtorName xtor) (dualPrdCnsType pol <$> lctx)
 
+
 dualPrdCnsType :: PolarityRep pol -> RST.PrdCnsType pol -> RST.PrdCnsType pol
-dualPrdCnsType PosRep (RST.PrdCnsType PrdRep ty) = RST.PrdCnsType CnsRep (dualType' PrdRep ty)
-dualPrdCnsType NegRep (RST.PrdCnsType PrdRep ty) = RST.PrdCnsType CnsRep (dualType' CnsRep ty)
-dualPrdCnsType PosRep (RST.PrdCnsType CnsRep ty) = RST.PrdCnsType PrdRep (dualType' CnsRep ty)
-dualPrdCnsType NegRep (RST.PrdCnsType CnsRep ty) = RST.PrdCnsType PrdRep (dualType' PrdRep ty)
+dualPrdCnsType PosRep (RST.PrdCnsType PrdRep ty) = RST.PrdCnsType CnsRep (embedTSTType (dualType' PrdRep (checkKind ty)))
```

Comment:
Wieso machst du hier den roundtrip über checkKind? Das wird im nächsten Schritt wahrscheinlich nicht mehr funktionieren, da `checkKind` dann wahrscheinlich in der Typechecker Monade laufen muss.

Reply:
weil dualType' in Dualize/Terms auch verwendet wird, und da auf TST Types arbeitet
nur in Dualize/Program wird es aus RST Typen angewendet